features/237-cms-validation/api.md

```diff
@@ -211,3 +214,105 @@ validations:
     min: 2000-01-01
     max: 2001-01-01
 ```
+
+### Null value handling
+
+Cases where `null` value appears:
+
+- value cleared for `Datetime`, `Reference`, `EmbeddedReference`, `Asset` field
+- value was originally `null` and not updated when save
+
+They may produce error when calling predefined functions, since `null` is a different data type from its original data type of the field type.
+
+Null value can be eliminated by adding extra checking in the expression, but this checking will be duplicated in most expression and may complicate the expression, which makes it hard to maintain.
+
+```yaml
+# bad example
+# Error is thrown if value == null, because length requires a string or array
+- expression: length(value) > 10
+
+# example
+- expression: value == null or length(value) > 10
+```
+
+`when` config is introduced to simplify the expression.
+
+- accepts an string expression which evaluates to give boolean value, like `expression`.
+- the effective boolean expression would be: `not (when) or (expression)`
+- `when` is optional, default `when` expression: `true`
```

Comment:
Does it contradict with the statement "Thus we provide a default when config, value != null, for all validations except required." below?

Reply:
I think validation can be a generic config across the whole cms, e.g. maybe later some config in user management also support validation. So i write this as a general description.

While currently we only support field validation and form validation, field validation is an exception for this.

Reply:
Maybe we can mention field validation has different default here to avoid confusion?